background.js

```diff
@@ -139,37 +139,49 @@ function main() {
         return courseList;
     }
 
+    async function getCourseName(courseCode, session, retries=0) {
+        const iframe = document.querySelector("#iframe-main").contentWindow.document;
+        const GRADES_API_URL = "https://ubcgrades.com/api/v2/grades/UBCV";
+        const STATISTICS_API_URL = "https://ubcgrades.com/api/v2/course-statistics/UBCV";
+        var completeURL;
+
+        if (retries <= 1) {
+            completeURL = GRADES_API_URL + '/' + session + '/' + courseCode[0] + '/' + courseCode[1] + '/OVERALL';
+        } else if (retries == 2)  {
+            console.log("Warning: " + courseCode + " does not seem to have a grade distribution, falling back to using course statistics");
+            completeURL = STATISTICS_API_URL + '/' + courseCode[0] + '/' + courseCode[1];
+        } else {
+            console.log("Error: Course name for " + courseCode + " not found");
+            return;
+        }
+
+        const response = await fetchResource(completeURL, { method: "GET", });
+        const data = await response.text();
+        let obj = data ? await JSON.parse(data) : {}
+        console.log(completeURL);
+        console.log(data);
+        if (obj.error == "Not Found") {
+            var sess_year = parseInt(session.slice(0, 4));
+            var sess_season = session.slice(-1);
+            var sess_last_year = (sess_year - 1).toString() + sess_season;
+            console.log("Error: data for " + courseCode + " during " + session + " unavailable, trying " + sess_last_year);
+            getCourseName(courseCode, sess_last_year, retries + 1);

```

Comment:
UBC Grades sometimes doesn't have data for the current year, try the last year as a fallback